components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/AWSLambdaClassMediator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.gateway.mediators;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.lambda.AWSLambda;
+import com.amazonaws.services.lambda.AWSLambdaClientBuilder;
+import com.amazonaws.services.lambda.model.InvocationType;
+import com.amazonaws.services.lambda.model.InvokeRequest;
+import com.amazonaws.services.lambda.model.InvokeResult;
+import org.apache.commons.lang.StringUtils;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.core.util.CryptoException;
+import org.wso2.carbon.core.util.CryptoUtil;
+import java.io.ByteArrayInputStream;
+import java.io.UnsupportedEncodingException;
+
+public class AWSLambdaClassMediator extends AbstractMediator {
+    private static final Log log = LogFactory.getLog(AWSLambdaClassMediator.class);
+    private String accessKey = "";
+    private String secretKey = "";
+    private String resourceName = "";
+
+    public AWSLambdaClassMediator() {
+
+    }
+
+    /**
+     * mediate to AWS Lambda
+     * @param messageContext - contains the payload
+     * @return true
+     */
+    public boolean mediate(MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+
+        String payload;
+        if (JsonUtil.hasAJsonPayload(axis2MessageContext)) {
+            payload = JsonUtil.jsonPayloadToString(axis2MessageContext);
+        } else {
+            payload = "{}";
+        }
+
+        InvokeResult invokeResult = invokeLambda(payload);
```

Comment:
What is the end user error message when the provided AWS credentials are incorrect or any other exception @ invokelamda?

Reply:
400 Bad Request is sent to the user